scalableminds/webknossos
# Add extended keyboard shortcut mode via ctrl + k for tool shortcuts

This PR adds a new "extended" mode to the keyboard shortcuts. This mode is activated using `ctrl + k` and gives access to more commands. Only shortcuts for switching between annotation tools are implemented, but more can be added. 
To leave the "extended" mode, the user needs to use one of the supported extended shortcuts or use `ctrl + k` again or wait 3 seconds.

Here is a list of all shortcuts for all tools. These shortcuts only work, once `ctrl + k` was pressed.
| Shortcut | Tool |
|---|---------------------|
| m | Move Tool           |
| s | Skeleton Tool       |
| b | Brush Tool          |
| e | Brush Erase Tool    |
| l | Lasso Tool          |
| r | Lasso lErase Too    |
| p | Segment Picker Tool |
| q | Quick Select Tool   |
| x | Bounding Box Tool   |
| o | Proofreading Tool   |

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- Open any (hybrid) tracing
- Press `ctrl + k` and try to quick-access the tools above by using their respective shortcuts. This should work.

### Issues:
- fixes #7045

### Open Questions:
- [ ] Where should I add the shortcut for the proofreading tool? Into the Skeleton or volume or default extended shortcuts? Additionally, do I need to make the proofreading tool shortcut only available if the proofreading tool is allowed? Because the shortcut can still exist without any problems and will not activate the proofreading tool, in case it is disabled for the user / lab / instance / dataset e.t.c.

### Known Problems:
- One should be able to deactivate the extended mode by pressing `ctrl + k` again, but Firefox does not trigger the second press as a key event and instead opens the Google search bar.

### TODOs:
- [x] Write changelog entry and link keyboard shortcuts doc.

------
(Please delete unneeded items, merge only when none are left open)
